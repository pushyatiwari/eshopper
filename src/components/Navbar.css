.header {
    display: flex;
    justify-content: space-between;
}

.navOrderDetails{
    display: flex;
    align-content: flex-end;
}