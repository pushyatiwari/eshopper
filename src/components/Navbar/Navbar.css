.header {
    display: flex;
    justify-content: space-between;
    background-color: rgb(70, 77, 77);
    text-decoration: none;
    color: white;
}

.nav-order-details{
    display: flex;
    align-content: flex-end;
    color: white;
    justify-content: space-between;
    margin-top: 12px;
}
button{
    color: white;
    text-decoration: none;
    background-color: transparent;  
}
.e-shopper{
    text-decoration: none;
    color: white;
}