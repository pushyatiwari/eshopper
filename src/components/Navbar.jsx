/* eslint-disable react/destructuring-assignment */
/* eslint-disable react/prop-types */
/* eslint-disable react/button-has-type */
import React from 'react';
import './Navbar.css';

const Navbar = (props) => (
  <header className="header">
    <p>E-shopper</p>
    <div className="navOrderDetails">
      <span>All Orders</span>
      <a href="/">
        My Basket :
        {props.cartCount}
      </a>
    </div>
  </header>
);

export default Navbar;
// https://5.imimg.com/data5/EX/QK/MY-37427162/selection_008-500x500.png
