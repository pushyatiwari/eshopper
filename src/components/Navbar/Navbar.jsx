/* eslint-disable react/button-has-type */
import React, { useContext } from 'react';
import './Navbar.scss';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import { ThemeContext } from '../../themeContext';

const Navbar = ({ cartCount }) => {
  const theme = useContext(ThemeContext);
  return (
    <header className={theme === 'dark' ? 'darkTheme header' : 'whiteTheme header'}>
      <Link to="/" className="e-shopper">
        <p>E-shopper</p>
      </Link>
      <div className="nav-order-details">
        <Link to="/allorders">
          <button className="nav-buttons">All Orders</button>
        </Link>
        <Link to="/cart" data-testid="cart-test">
          <button className={theme === 'dark' ? 'darkTheme nav-buttons' : 'whiteTheme nav-buttons'}>
            My Basket :
            {cartCount}
          </button>
        </Link>
      </div>
    </header>
  );
};

Navbar.propTypes = {
  cartCount: PropTypes.number.isRequired,

};

export default Navbar;
// https://5.imimg.com/data5/EX/QK/MY-37427162/selection_008-500x500.png
