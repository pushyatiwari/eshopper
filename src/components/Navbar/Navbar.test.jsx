import {
  fireEvent,
  render, screen,
} from '@testing-library/react';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import Navbar from './Navbar';

describe(Navbar.name, () => {
  test('should display para', () => {
    render(
      <BrowserRouter>
        <Navbar
          cartCount={2}
        />
      </BrowserRouter>,
    );
    const navbarElement = screen.getByText('E-shopper');
    expect(navbarElement.tagName).toBe('P');
  });
  test('should display all orders', () => {
    render(
      <BrowserRouter>
        <Navbar
          cartCount={2}
        />
      </BrowserRouter>,
    );
    const navbarElement = screen.getByText('All Orders');
    expect(navbarElement.tagName).toBe('BUTTON');
  });
  test('should render the basket', () => {
    render(
      <BrowserRouter>
        <Navbar
          cartCount={2}
        />
      </BrowserRouter>,
    );
    const cart = screen.getByTestId('cart-test');
    fireEvent.click(cart);
    expect(document.location.href).toBe('http://localhost/cart');
  });
  test('should be called with', () => {
    const navbarContainer = render(
      <BrowserRouter>
        <Navbar
          cartCount={2}
        />
      </BrowserRouter>,
    );
    expect(navbarContainer).toMatchSnapshot();
  });
});
