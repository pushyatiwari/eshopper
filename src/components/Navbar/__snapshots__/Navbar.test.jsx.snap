// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should be called with 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="whiteTheme header"
      >
        <a
          class="e-shopper"
          href="/"
        >
          <p>
            E-shopper
          </p>
        </a>
        <div
          class="nav-order-details"
        >
          <a
            href="/allorders"
          >
            <button
              class="nav-buttons"
            >
              All Orders
            </button>
          </a>
          <a
            data-testid="cart-test"
            href="/cart"
          >
            <button
              class="whiteTheme nav-buttons"
            >
              My Basket :
              2
            </button>
          </a>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="whiteTheme header"
    >
      <a
        class="e-shopper"
        href="/"
      >
        <p>
          E-shopper
        </p>
      </a>
      <div
        class="nav-order-details"
      >
        <a
          href="/allorders"
        >
          <button
            class="nav-buttons"
          >
            All Orders
          </button>
        </a>
        <a
          data-testid="cart-test"
          href="/cart"
        >
          <button
            class="whiteTheme nav-buttons"
          >
            My Basket :
            2
          </button>
        </a>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
